reframe/frontend/executors/__init__.py

```diff
@@ -763,9 +815,42 @@ def timeout_expired(self):
     def enter(self):
         self._num_failed_tasks = 0
 
-    def exit(self):
+    def _exit(self):
         pass
 
     @abc.abstractmethod
-    def runcase(self, case):
-        '''Run a test case.'''
+    def execute(self, testcases):
+        '''Execute the policy for a given set of testcases and exit.'''
+
+
+def asyncio_run(coro):
```

Comment:
Once I have tried an async def of the `main()` and started the loop there with `asyncio.run(main())`. I haven't changed anything else in the code and that was enough to run as-is the existing code (ie. not asyncio-based). Why do we need all this complex event loop logic here?